vault/resource_aws_secrets_sync_destination.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const (
+	fieldAccessKeyID     = "access_key_id"
+	fieldSecretAccessKey = "secret_access_key"
+)
+
+var awsSyncDestinationFields = []string{
+	fieldAccessKeyID,
+	fieldSecretAccessKey,
+	consts.FieldAWSRegion,
+}
+
+func awsSecretsSyncDestinationResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(awsSecretsSyncDestinationWrite, provider.VaultVersion115),
+		ReadContext:   provider.ReadContextWrapper(awsSecretsSyncDestinationRead),
+		DeleteContext: awsSecretsSyncDestinationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name of the AWS destination.",
+				ForceNew:    true,
+			},
+			fieldAccessKeyID: {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
Do GH, Azure and Vercel also allow things to be set via the environment? From the documentation, I got the idea that it was only AWS and GCP that have enabled setting values through environment variables.

I did not adhere to this behavior in GCP though, so your point about the mismatch still stands. If Vault is already performing validations for certain fields via environment variables, I would prefer to let Vault handle the validation and not have the fields be `Required` in TFVP. This would allow users to configure fields via the environment for Vault if they'd like and leave the fields blank in their TF configs. Definitely open to thoughts and suggestions here on what would make a better UX for TFVP users

Reply:
I'm drawing a blank at the moment to say for sure, but I thought all the store types (IDK about Vercel) can be configured through the environment.

Reply:
[Here is a doc with more context](https://docs.google.com/document/d/11O1TJiCmHx2CSwQ6lHxDtEMQVISvec_P_FhxLBBWeE4/edit#heading=h.d8142o1blz9t) on which attributes are editable and which can be provided via environment variables.

Only Vercel doesn't have the option to use the environment at the moment but we're thinking of adding that capability for consistency as part of [this ticket](https://hashicorp.atlassian.net/browse/VAULT-21418).

Reply:
Thanks for the context! Updated fields that are available via environment variables to be `Optional`. I will document that they can be left omitted and provided directly to Vault using environment variables using info from the context sheet linked above 🙏🏼 

For this PR, I have decided to follow TFVP's precedent to let Vault handle environment variable reads for credentials in case the values are left omitted in the TFVP config. This would mean that TFVP itself is not going to read credentials from the environment and pass them to Vault, but would instead pass an empty string to Vault and rely on Vault's internal mechanisms to read the environment variables

I went with this approach in order to keep the interface for Secrets Sync resources simple, but can add in enhancements to read environment variables also on the TFVP side if folks think that that is necessary for the flow. Also happy to answer any questions on this!